backend/src/database/models.rs

```diff
@@ -619,9 +633,24 @@ impl Role {
 
         Some(())
     }
+
+    pub fn delete_deep(conn: &PgConnection, role_id: i32) -> Option<()> {
```

Comment:
I have a suspicion this is because my own `delete_deep` code returns `Option<()>` but this should really be a `Result` to be more idiomatic (ie, types provide context information, delete_deep should return the result of did the operation go successfully `Ok` or was it an error? `Err`)